### DISCUSSION TITLE
a idea about remove bridge in dart model

### DISCUSSION THREAD

**Discussion by huang12zheng** [fzyzcjy/flutter_rust_bridge#1166] [Ideas]
```dart
class User {
  final FlutterRustBridgeExample bridge; // remove it
  final String name;
  final int id;

  const User({
    required this.bridge,
    required this.name,
    required this.id,
  });
```
* ffi_io.dart
```dart
late final api = FlutterRustBridgeExampleImpl(dylib);
```
From the look of these two codes,
we cloud use api instead of bridge.

What do you think about that?
> 1. By the way, I think it will also hidden #969
> 2. Due to `pub rust_input: Vec<String>,` , I would use that variable as api's name.
> 3. Maybe you could add a switch to define whether it's on or off, but I don't think so

**Comment by Desdaemon** [comment]
Interesting idea. I've also been encountering this issue when writing my own bindings, but came to the conclusion that this will only make sense for end-users of a library. It could be another option within codegen, and should be a relatively straightforward addition to the library.
